samples/key-value-customer-registry/src/main/java/customer/api/CustomerEndpoint.java

```diff
@@ -108,32 +108,63 @@ public HttpResponse getCustomersCsvByName(String name) {
         .stream(CustomersByName::continuousGetCustomerSummaryStream)
         .source(name);
 
-    final var eventPrefix = ByteString.fromString("data: ");
-    final var eventEnd = ByteString.fromString("\n\n");
-    // Server sent events
-    // https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event_stream_format
-    Source<ByteString, NotUsed> sseCustomerSummaryStream =
-        customerSummarySource.map(customerSummary ->
-            eventPrefix.concat(JsonSupport.encodeToAkkaByteString(customerSummary)).concat(eventEnd)
-        );
-
-    return HttpResponse.create()
-        .withStatus(StatusCodes.OK)
-        .withHeaders(Arrays.asList(
-            CacheControl.create(CacheDirectives.NO_CACHE),
-            Connection.create("keep-alive")
-        ))
-        .withEntity(HttpEntities.create(TEXT_EVENT_STREAM, sseCustomerSummaryStream));
+    return HttpResponses.serverSentEvents(customerSummarySource); // <2>
+  }
+  // end::sse-view-updates[]
+
+  // tag::sse-customer-changes[]
+  private record CustomerStreamState(Optional<Customer> customer, boolean isSame) {}
+
+  @Get("/stream-customer-changes/{customerId}")
+  public HttpResponse streamCustomerChanges(String customerId) {
+    Source<Customer, Cancellable> stateEvery5Seconds =
+        // stream of ticks, one immediately, then one every five seconds
+        Source.tick(Duration.ZERO, Duration.ofSeconds(5), "tick") // <1>
+          // for each tick, request the entity state
+          .mapAsync(1, __ ->
+            componentClient.forKeyValueEntity(customerId)
+                .method(CustomerEntity::getCustomer)
+                .invokeAsync().handle((Customer customer, Throwable error) -> {
+                  if (error == null) {
+                    return Optional.of(customer);
+                  } else if (error instanceof IllegalArgumentException) {
+                    // calling getCustomer throws IllegalArgument if the customer does not exist
+                    // we want the stream to continue polling in that case, so turn it into an empty optional
+                    return Optional.<Customer>empty();
+                  } else {
+                    throw new RuntimeException("Unexpected error polling customer state", error);
+                  }
+                })
+          )
+          // then filter out the empty optionals and return the actual customer states for nonempty
+          // so that the stream contains only Customer elements
+          .filter(Optional::isPresent).map(Optional::get);
+
+    // deduplicate, so that we don't emit if the state did not change from last time
+    Source<Customer, Cancellable> streamOfChanges = // <2>
+        stateEvery5Seconds.scan(new CustomerStreamState(Optional.empty(), true),
+          (state, newCustomer) ->
+            new CustomerStreamState(Optional.of(newCustomer), state.customer.isPresent() && state.customer.get().equals(newCustomer))
```

Comment:
or
```
new CustomerStreamState(Optional.of(newCustomer), state.customer.equals(Option.of(newCustomer))
```

In my example I had made the first `stateEvery5Seconds` a `Source<Option<Customer>`

but not very important